core/import-shearwater.c

```diff
@@ -97,8 +98,26 @@ extern int shearwater_profile_sample(void *handle, int columns, char **data, cha
 		cur_sample->ndl.seconds = atoi(data[4]) * 60;
 	if (data[5])
 		cur_sample->cns = atoi(data[5]);
-	if (data[6])
-		cur_sample->stopdepth.mm = metric ? atoi(data[6]) * 1000 : feet_to_mm(atoi(data[6]));
+	if (data[6]) {
+		d6 = atoi(data[6]);
+		if (d6 > 0) {
+			cur_sample->stopdepth.mm = metric ? d6 * 1000 : feet_to_mm(d6);
+			cur_sample->in_deco = 1;
+		} else if (data[7]) {
+			d7 = atoi(data[7]);
+			if (d7 > 0) {
+				cur_sample->stopdepth.mm = metric ? d7 * 1000 : feet_to_mm(d7);
+				if (data[8]) {
+					cur_sample->stoptime.seconds = atoi(data[8]) * 60;
+				}
```

Comment:
[1] 

> With the change to avoid multiple atoi calls, this is now needed :D

hmm, the `{ }` are not needed here.

Reply:
ah, on the quoted email it shows quite a different code snippet and thus I missed you meant the innermost if